src/Controller/OcrController.php

```diff
@@ -108,31 +108,57 @@ public function getLangs(Request $request): array
      * @Route("/", name="home")
      * @return Response
      */
-    public function home(): Response
+    public function homeAction(): Response
     {
         if ($this->imageUrl) {
             static::$params['text'] = $this->engine->getText($this->imageUrl, static::$params['langs']);
         }
 
+        // Pre-supply available langs for autocompletion in the form.
+        static::$params['available_langs'] = $this->engine->getValidLangs();
+        sort(static::$params['available_langs']);
+
         return $this->render('output.html.twig', static::$params);
     }
 
     /**
-     * @Route("/api.php", name="api")
+     * @Route("/api", name="api")
+     * @Route("/api.php", name="apiPhp")
+     * @return JsonResponse
+     */
+    public function apiAction(): JsonResponse
+    {
+        return $this->getApiResponse([
+            'text' => $this->engine->getText($this->imageUrl, static::$params['langs']),
+        ]);
+    }
+
+    /**
+     * @Route("/api/available_langs", name="apiLangs")
      * @return JsonResponse
      */
-    public function api(): JsonResponse
+    public function apiAvailableLangsAction(): JsonResponse
+    {
+        return $this->getApiResponse([
+            'available_langs' => $this->engine->getValidLangs(),
```

Comment:
This consequently includes a bunch of irrelevant things in the response like the Tesseract options, and `langs` which is actually kind of confusing. I'm assuming it doesn't matter much, but if it does we can `unset` those keys just for this API endpoint. Or even return just the languages as an array, and nothing else?

Reply:
I don't think it matters too much, and might be useful if we do any normalization of language codes, e.g. requesting `/api/available_langs?langs[]=Foo-bar` returns `{available_langs:{…}, langs: ['foo_bar']}` or whatever?

Are we going to translate language names anywhere? Because then sending `?uselang=x` would change the result.

Reply:
I changed it to return only `available_langs` and the request engine (since if you don't request an engine, it defaults to `google` so we should let them know that)

Reply:
:+1:

Reply:
I commented before seeing your reply... I see what you're saying about language code normalization! Yes that's a good idea, but should /api/available_langs be responsible for that? Either way probably better for a separate PR?

Reply:
Yeah I think it can wait. There's a whole bunch of stuff about what the wiki-side UI is going to need that doesn't feel fully figured out yet (at least to me). :)